library/src/main/java/com/pokegoapi/auth/PtcCredentialProvider.java

```diff
@@ -227,72 +231,30 @@ private void login(String username, String password, int attempt)
 					throw new InvalidCredentialsException(builder.toString());
 				}
 			}
-
-			String ticket = null;
-			for (String location : response.headers("location")) {
-				String[] ticketArray = location.split("ticket=");
-				if (ticketArray.length > 1) {
-					ticket = ticketArray[1];
-				}
-			}
-
-			if (ticket == null) {
-				throw new LoginFailedException("Failed to fetch token, body:" + body);
-			}
-
-			url = HttpUrl.parse(LOGIN_OAUTH).newBuilder()
-					.addQueryParameter("client_id", CLIENT_ID)
-					.addQueryParameter("redirect_uri", REDIRECT_URI)
-					.addQueryParameter("client_secret", CLIENT_SECRET)
-					.addQueryParameter("grant_type", "refreshToken")
-					.addQueryParameter("code", ticket)
-					.build();
-
-			postRequest = new Request.Builder()
-					.url(url)
-					.method("POST", reqBody)
-					.build();
-
-			try {
-				response = client.newCall(postRequest).execute();
-			} catch (IOException e) {
-				throw new LoginFailedException("Network Failure ", e);
-			}
-
-			try {
-				body = response.body().string();
-			} catch (IOException e) {
-				throw new LoginFailedException("Network failure", e);
-			}
-
-			String[] params;
-			try {
-				params = body.split("&");
-				int expire = Integer.valueOf(params[1].split("=")[1]);
-				tokenId = params[0].split("=")[1];
-				expiresTimestamp = time.currentTimeMillis() + (expire * 1000 - REFRESH_TOKEN_BUFFER_TIME);
-				unknown2 = expire;
-				if (random.nextDouble() > 0.1) {
-					unknown2 = UK2_VALUES[random.nextInt(UK2_VALUES.length)];
+			List<Cookie>c = client.cookieJar().loadForRequest(localObject);
```

Comment:
There should be a space before the variable name here. Also, a more descriptive name would be useful.